### ISSUE TITLE
Support TLS in kafka output testing

### ISSUE THREAD

**Issue by jsoriano** [elastic/beats#10989]
#10834 added supoort for kafka output testing, but it doesn't support TLS. Add support for TLS too.

**Comment by faec** [comment]
Completed in https://github.com/elastic/beats/pull/28971

**Comment by joostdecock** [comment]
@faec You closed this as completed, but #28971 adds support for SASL/SCRAM, not for TLS.

Is it possible that was a mixup, because TLS is (still) not supported in the output test:

```
TLS... WARN Kafka output doesn't support TLS testing
Kafka: poc-morio-node1.cert.europa.eu:9092...
  parse host... OK
  dns lookup... OK
  addresses: 127.0.1.1
  dial up... OK
```

As someone using mTLS for kafka authentication, wrongly configured certificates are a very common reason for things not working as expected, and it would be incredibly helpful if the tests could verify the TLS connection.

As such, could we perhaps re-open this issue? Because while #28971 was a great contribution, I don't think it's actually addresses this particular issue at all.
